engine/include/Engine/Intermediate/TransformComponent.hpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Intermediate/Component.hpp"
+#include "Engine/Intermediate/GameObject.hpp"
+#include "GPM/Conversion.hpp"
+#include "GPM/Matrix4.hpp"
+#include "GPM/Vector3.hpp"
+#include "GPM/Vector4.hpp"
+#include "GPM/Quaternion.hpp"
+#include "GPM/Transform.hpp"
+#include <vector>
+
+namespace GPE
+{
+class TransformComponent : public Component
+{
+public:
+    struct CreateArg
+    {
+        GPM::Vec3 position      = GPM::Vec3::zero();
+        GPM::Vec3 eulerRotation = GPM::Vec3::zero();
+        GPM::Vec3 scale         = GPM::Vec3::one();
+    };
+
+protected:
+    GPM::SplitTransform m_spaceAttribut;
```

Comment:
If you want to keep the `GPM::SplitTransform`:
```suggestion
    GPM::SplitTransform m_spaceAttribute; // or m_splitTransform?
```